app/scripts/migrations/107.ts

```diff
@@ -0,0 +1,82 @@
+import { cloneDeep } from 'lodash';
+import { hasProperty, isObject } from '@metamask/utils';
+
+export const version = 107;
+
+interface Account {
+  address: string;
+  balance: string;
+}
+
+interface AccountTrackerControllerState {
+  accountsByChainId: Record<string, Record<string, Account>>;
+}
+
+/**
+ * Migrates state from the now removed CachedBalancesController to the AccountTrackerController and formats it accordingly.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(originalVersionedData: {
+  meta: { version: number };
+  data: Record<string, unknown>;
+}) {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  versionedData.data = transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>) {
+  if (
+    !hasProperty(state, 'CachedBalancesController') ||
+    !isObject(state.CachedBalancesController) ||
+    !hasProperty(state.CachedBalancesController, 'cachedBalances') ||
+    !isObject(state.CachedBalancesController.cachedBalances) ||
+    !hasProperty(state, 'AccountTracker')
+  ) {
+    return state;
+  }
+
+  const accountTrackerControllerState = (state.AccountTracker ||
+    {}) as AccountTrackerControllerState;
+
+  if (!accountTrackerControllerState.accountsByChainId) {
+    accountTrackerControllerState.accountsByChainId = {};
+  }
+
+  const cachedBalances = state.CachedBalancesController
+    .cachedBalances as Record<string, Record<string, string>>;
+
+  Object.keys(cachedBalances).forEach((chainId) => {
+    if (!accountTrackerControllerState.accountsByChainId[chainId]) {
+      accountTrackerControllerState.accountsByChainId[chainId] = {};
+    }
+
+    Object.keys(cachedBalances[chainId]).forEach((accountAddress) => {
+      const balance = cachedBalances[chainId][accountAddress];
+      if (
+        accountTrackerControllerState.accountsByChainId[chainId][
+          accountAddress
+        ] === undefined
+      ) {
+        accountTrackerControllerState.accountsByChainId[chainId][
+          accountAddress
+        ] = {
+          address: accountAddress,
```

Comment:
looks like address isn't defined in the AccountTrackerController state

https://github.com/MetaMask/core/blob/76787ceeab5083780f50a2f9e68617f08cc600c4/packages/assets-controllers/src/AccountTrackerController.ts#L29

Reply:
wrong AccountTrackerController ☝️  🥲

Reply:
asdfjlk;. Carry on